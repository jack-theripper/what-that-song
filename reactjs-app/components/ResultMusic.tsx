import React, {useMemo} from 'react';
import {TMusic} from "../types";

interface ResultMusicProps {
    item: TMusic;
}

const ResultMusic: React.FC<ResultMusicProps> = ({item}) => {

    const artist = useMemo<string>(() => item.artists.map(artist => artist.name).join(', '), [item.artists]);

    return (
        <div>
            <h3>{artist.length > 0 && `${artist} - `}{item.title} (совпадение {item.score}%)</h3>
            <ul>
                <li>Альбом: {item.album?.name}</li>
                <li>Жанры: {item.genres?.map(genre => (<span key={genre.name}>{genre.name}</span>))}</li>
                <li>Лэйбл: {item.label}</li>
                <li>Дата релиза: {item.release_date}</li>

                {item.external_metadata && Object.keys(item.external_metadata).map(externalId => (
                    (externalId === 'youtube') && (<li key={externalId}>https://www.youtube.com/watch?v={item.external_metadata?.youtube?.vid}</li>)
                ))}

                {item.external_metadata?.spotify && (
                    <li>https://open.spotify.com/track/{item.external_metadata.spotify.track.id}</li>
                )}

                {item.external_metadata?.musicbrainz && (
                    item.external_metadata.musicbrainz.map(item => (
                        <li>https://musicbrainz.org/recording/{item.track.id}</li>
                    ))
                )}



            </ul>


        </div>
    );
};

export default ResultMusic;