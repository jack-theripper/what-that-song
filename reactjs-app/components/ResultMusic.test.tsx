import ResultMusic from "./ResultMusic";
import renderer from 'react-test-renderer';

const value = {
    "acrid": "b5f885d01752738bdec0a3b4077dcb23",
    "external_ids": {
        "isrc": "RUA3R2141382",
        "upc": "7422400001020"
    },
    "artists": [
        {
            "name": "LAZZY2WICE"
        }
    ],
    "result_from": 1,
    "title": "Do You Pray?",
    "play_offset_ms": 14580,
    "score": 100,
    "label": "MERLIN - LazzyMusic",
    "external_metadata": {
        "deezer": {
            "artists": [
                {
                    "name": "LAZZY2WICE"
                }
            ],
            "album": {
                "name": "The Blind Victim"
            },
            "track": {
                "name": "Do You Pray?",
                "id": "1825985157"
            }
        }
    },
    "album": {
        "name": "The Blind Victim"
    },
    "duration_ms": 122000,
    "release_date": "2022-07-15"
};

describe('ResultMusic component', () => {

    test('it rendering', () => {
        const component = renderer.create(<ResultMusic item={value}/>);
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });

})