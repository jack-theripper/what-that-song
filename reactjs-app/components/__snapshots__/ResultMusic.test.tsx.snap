// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultMusic component it rendering 1`] = `
<div>
  <h3>
    LAZZY2WICE - 
    Do You Pray?
     (совпадение 
    100
    %)
  </h3>
  <ul>
    <li>
      Альбом: 
      The Blind Victim
    </li>
    <li>
      Жанры: 
    </li>
    <li>
      Лэйбл: 
      MERLIN - LazzyMusic
    </li>
    <li>
      Дата релиза: 
      2022-07-15
    </li>
  </ul>
</div>
`;
